File: py-game/game/objects/bomb.py
from config import COUNTDOWN_BOMB_MS
from game.objects.base import BaseObj


class BombObj(BaseObj):
    def __init__(self, index_x: int, index_y: int, power: int, man_key: str):
        super().__init__(index_x, index_y, "bomb")
        self.power = power
        self.man_key = man_key
        self.remaining_ms = COUNTDOWN_BOMB_MS

File: backend2/game/objects/bomb.go
package objects

import (
	"bomberman/backend2/config"
	"bomberman/backend2/types"
)

type BombObj struct {
	Base        BaseObj
	RemainingMs uint32
	Power       uint32
	ManKey      types.ManSpriteKey
}

func NewBombObj(index *types.MapIndex, power uint32, manKey types.ManSpriteKey) *BombObj {
	return &BombObj{
		Base:        NewBaseObj(index, types.ObjBomb),
		RemainingMs: config.BombCountdownMs,
		Power:       power,
		ManKey:      manKey,
	}
}
